packages/@aws-cdk/aws-lambda/lib/code.ts

```diff
@@ -36,7 +36,10 @@ export abstract class Code {
   }
 
   /**
-   * Loads the function code from a local disk asset.
+   * Loads the function code from a local disk path.
+   *
+   * This is a convenience wrapper around fromAssetObject(new s3_assets.Asset(...)).
+   *
```

Comment:
Yeah, I thought I would "just" make that change immediately as well, so we can do something that seems like it should be trivially possible like `lambda.Code.fromAssetObject(new MyCompanysInternalAssetSource('MyPackage'));`.

But it's hard because Asset objects are constructs and so it's not so easy to make it all work nicely.